test/browser/keyboardNavigation.test.tsx

```diff
@@ -28,161 +27,165 @@ const columns = [
-  validateCellPositionOld(0, 0);
+  validateCellPosition(0, 0);
 
   // tab at the end of a row selects the first cell on the next row
   await userEvent.keyboard('{end}');
   await userEvent.tab();
-  validateCellPositionOld(0, 1);
+  validateCellPosition(0, 1);
 
   // shift tab should select the last cell of the previous row
   await userEvent.tab({ shift: true });
-  validateCellPositionOld(6, 0);
+  validateCellPosition(6, 0);
 });
 
 test('arrow and tab navigation', async () => {
-  setupOld({ columns, rows, bottomSummaryRows });
+  setup({ columns, rows, bottomSummaryRows });
 
   // pressing arrowleft on the leftmost cell does nothing
   await userEvent.tab();
   await userEvent.keyboard('{arrowdown}');
-  validateCellPositionOld(0, 1);
+  validateCellPosition(0, 1);
   await userEvent.keyboard('{arrowleft}');
-  validateCellPositionOld(0, 1);
+  validateCellPosition(0, 1);
 
   // pressing arrowright on the rightmost cell does nothing
   await userEvent.keyboard('{end}');
-  validateCellPositionOld(6, 1);
+  validateCellPosition(6, 1);
   await userEvent.keyboard('{arrowright}');
-  validateCellPositionOld(6, 1);
+  validateCellPosition(6, 1);
 
   // pressing tab on the rightmost cell navigates to the leftmost cell on the next row
   await userEvent.tab();
-  validateCellPositionOld(0, 2);
+  validateCellPosition(0, 2);
 
   // pressing shift+tab on the leftmost cell navigates to the rightmost cell on the previous row
   await userEvent.tab({ shift: true });
-  validateCellPositionOld(6, 1);
+  validateCellPosition(6, 1);
 });
 
 test('grid enter/exit', async () => {
-  setupOld({ columns, rows: new Array(5), bottomSummaryRows });
+  setup({ columns, rows: new Array(5), bottomSummaryRows });
 
   // no initial selection
-  expect(getSelectedCellOld()).not.toBeInTheDocument();
+  await expect.element(getSelectedCell()).not.toBeInTheDocument();
 
   // tab into the grid
   await userEvent.tab();
-  validateCellPositionOld(0, 0);
+  validateCellPosition(0, 0);
 
   // shift+tab tabs out of the grid if we are at the first cell
   await userEvent.tab({ shift: true });
   expect(document.body).toHaveFocus();
 
   await userEvent.tab();
-  validateCellPositionOld(0, 0);
+  validateCellPosition(0, 0);
 
   await userEvent.keyboard('{arrowdown}{arrowdown}');
-  validateCellPositionOld(0, 2);
+  validateCellPosition(0, 2);
 
   // tab should select the last selected cell
   // click outside the grid
   await userEvent.click(document.body);
   await userEvent.tab();
   await userEvent.keyboard('{arrowdown}');
-  validateCellPositionOld(0, 3);
+  validateCellPosition(0, 3);
 
   // shift+tab should select the last selected cell
   await userEvent.click(document.body);
+  // TODO: why do we need to tab three times?
   await userEvent.tab({ shift: true });
+  await userEvent.tab({ shift: true });
+  await userEvent.tab({ shift: true });
+  expect(document.body).not.toHaveFocus();
   await userEvent.keyboard('{arrowup}');
```

Comment:
Why do we press arrow up? Can we not get a selected cell without it?

Reply:
I think the idea is to ensure cell receives focus after shift + tab. `validateCellPosition` call will pass even if the cell does not have focus

Reply:
We could check that the selected cell has focus

Reply:
Done